api/envoy/extensions/transport_sockets/alts/v3/alts.proto

```diff
@@ -15,6 +15,7 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#extension: envoy.transport_sockets.alts]
 
 // Configuration for ALTS transport socket. This provides Google's ALTS protocol to Envoy.
+// Store the tsi peer name in dynamic metadata.
```

Comment:
I would also add the key name in the metadata and also document that you are storing Struct under that key with the "peer_identity" field, so people do not need to go hunting for it in the source code.